tracing/tracer.go

```diff
@@ -36,6 +37,21 @@ func init() {
 	opentracing.SetGlobalTracer(&globalTracer)
 	// Init the global allow-list
 	SetMetricsTagsAllowList(nil)
+	// Register a ValueWrap with the detach library
+	detach.Register(detach.Hooks{
+		Inline: func(dst, src context.Context) context.Context {
+			return opentracing.ContextWithSpan(dst, opentracing.SpanFromContext(src))
+		},
+		Async: func(dst, src context.Context, next func(ctx context.Context)) {
+			span, _ := opentracing.StartSpanFromContext(
```

Comment:
🔕 
1. we probably want to verify that our tracing system is ok with having a child span finishes after the parent span finishes
2. we probably want to check that there's actually a span in `src` before creating the child span? it's possible `detach.Async` is called from a context without any tracing, and we probably don't want to create this top-level span in those cases.